### PR TITLE
add @PactUrl to pact-jvm-provider-junit

### PR DESCRIPTION
**Pull request by stefan-lz**
Please review this and merge into master for me

Cheers,

### CODE REVIEW

**Comment by uglyog** [comment]
I've merged it, but I'm missing seeing the link between the annotation and the loader. What handles calling the loader when the annotation is present?

**Comment by idntfy** [comment]
That's [handed by infrastructure](https://github.com/DiUS/pact-jvm/blob/3_2_1/pact-jvm-provider-junit/src/main/java/au/com/dius/pact/provider/junit/PactRunner.java#L94) that has a pact source abstraction. PactUrl is just another pact source, like [PactBroker](https://github.com/DiUS/pact-jvm/blob/3_2_1/pact-jvm-provider-junit/src/main/java/au/com/dius/pact/provider/junit/loader/PactBroker.java) or [PactFolder](https://github.com/DiUS/pact-jvm/blob/3_2_1/pact-jvm-provider-junit/src/main/java/au/com/dius/pact/provider/junit/loader/PactFolder.java).
Basically anyone can add any pact source. For example, you can have even your company-specific `@MyCompanyPactSource` in order not to hardcode pact broker url everywhere.

**Comment by uglyog** [comment]
Nice!
